ability upgrade event

BLBLBI opened the pull request:
https://github.com/FreezingMoon/AncientBeast/issues/1243
Here is the ability upgrade event.
I found another way to do the fade-in/out than initially anticipated so i hope it's good enough

<a href="https://gitpod.io/#https://github.com/FreezingMoon/AncientBeast/pull/1685"><img src="https://gitpod.io/api/apps/github/pbs/github.com/BLBLBI/AncientBeast.git/646e88a1173be136f8676c74c5206a0a3aef6688.svg" /></a>

DreadKnight commented:
Pretty good. I see sound still missing, but I thought about it and you might as well remove that piece of code I guess, because it would be rather annoying and kinda interferes with the current action anyway..

Did a bit of testing and found a small issue: after an ability upgrades, the only frame is displayed on top of the upgraded frame.

BLBLBI commented:
Yeah sound is there but it's the stepping sound so with the music + attack sound you can't here it (you can without music) but yeah either remove it or maybe delay it a bit?.
Also can you rephrase the small issue please? I don't get the "the only frame" part (english is not my main language sorry)

DreadKnight commented:
Yeah, I guess a delay would really help, maybe we can try a second delay to start with, so that it won't overlap the ability sound.

This is what I mean: during the round an ability upgraded, there's the regular frame on top of the upgraded frame
![ab](https://user-images.githubusercontent.com/587487/85955449-53732500-b987-11ea-85bb-df16c3c5751d.jpg)

BLBLBI commented:
I added 300ms of delay and changed the sound so it's a bit more noticeable (it's the attack sound now or something close to it), 1s seems a bit too much but we can always change it if you prefer.
For the frame however, i looked in the files and it's the frame_upgrade_dimmed that you see, not the normal frame overriding the upgrade frame ( you can see it in assets/interface/frame_pgrade_dimmed.png).

DreadKnight commented:
@BLBLBI Understood, will test some more. Regarding sound, we should find and include something good for this,
quite a few sfx over on OGA, under free licenses, as we need them to be for the project https://opengameart.org

Regarding the frame, ewww! Fail, I thought that atrocity was done by stacking simpler frames on top of each other.
Will fix that stuff myself in a bit, already on it.

DreadKnight commented:
Alright, I've fixed the frames and I've found and committed a sfx for this in 52d6b75c019530009db75764f71bae1aaed7d602

BLBLBI commented:
And here is the ability upgrade with the new sound

DreadKnight commented:
Pretty good overall. Found some small kinks here and there, but will open up new issues for those and for things related to this.
